test/test_cost.py

```diff
@@ -155,6 +155,10 @@
 )
 
 
+# Increase recursion limit, some objectives have more than 1000 tests
+sys.setrecursionlimit(2000)
```

Comment:
I needed to increase this. Otherwise classes with more than 1000 tests fail after hitting this number